pinot-query-runtime/src/main/java/org/apache/pinot/query/service/server/QueryServer.java

```diff
@@ -163,37 +169,48 @@ public void submit(Worker.QueryRequest request, StreamObserver<Worker.QueryRespo
       return;
     }
 
-    try (QueryThreadContext.CloseableContext queryTlClosable = QueryThreadContext.openFromRequestMetadata(reqMetadata);
+    try (QueryThreadContext.CloseableContext qTlClosable = QueryThreadContext.openFromRequestMetadata(reqMetadata);
         QueryThreadContext.CloseableContext mseTlCloseable = MseWorkerThreadContext.open()) {
+
       long requestId = QueryThreadContext.getRequestId();
       QueryThreadContext.setQueryEngine("mse");
 
-      Tracing.ThreadAccountantOps.setupRunner(Long.toString(requestId), ThreadExecutionContext.TaskType.MSE);
-      ThreadExecutionContext parentContext = Tracing.getThreadAccountant().getThreadExecutionContext();
-      try {
-        forEachStage(request,
-            (stagePlan, workerMetadata) -> {
+      // Submit the stage for each worker
+      CompletableFuture<List<Object>>[] futures = forEachStageAndWorker(request,
+          (stagePlan, workerMetadata) -> {
+            Tracing.ThreadAccountantOps.setupRunner(Long.toString(requestId), ThreadExecutionContext.TaskType.MSE);
+            ThreadExecutionContext parentContext = Tracing.getThreadAccountant().getThreadExecutionContext();
+
+            try {
               _queryRunner.processQuery(workerMetadata, stagePlan, reqMetadata, parentContext);
```

Comment:
Seems the return value is ignored. Is this correct? Do we want to track the status of this future?